### ISSUE TITLE
Incompatibility between BloodArsenal and VoidMonster on servers

### ISSUE THREAD

**Issue by Mettymagic** [ReikaKalseki/Reika_Mods_Issues#1624]
I was getting an odd "A fatal error has occurred" kick when I tried to join a custom modpack server; when joining it, it would show the world for a split second, then kick me off. The game did not crash, meaning it was a server-side thing, so I decided to do some investigation as to what caused this. Looking in the client log revealed [this error.](https://hastebin.com/dayarodiba.css) Seeing as though it was an Entity ID mismatch, I used Anti ID Conflict to check the entity IDs on both the server and the client, which revealed a difference: the server list included BloodTNT but the client one did not. I created a new modpack with just DragonAPI, VoidMonster, BloodMagic, BloodArsenal, ExCore and AntiIDConflict and sure enough the error returned. The server worked fine with just BloodMagic/BloodArsenal. I also tried it without AntiIDConflict/ExCore and still got the error when the mods were paired.

So in conclusion, something with the EntityIDs messes up when the two are paired together, making servers unplayable due to this. I'm not sure what causes this specifically, but something prevents the EntityID of the BloodTNT from generating properly.

[Client Entity List](https://hastebin.com/ocujisixaz.css)
[Server Entity List](https://hastebin.com/itororuqeg.css)

**Comment by ReikaKalseki** [comment]
My registration code is [here](https://github.com/ReikaKalseki/VoidMonster/blob/master/VoidMonster.java#L92) and is normal. Is Blood TNT's entity ID hardcoded to some random value?

**Comment by Barhandar** [comment]
No. It uses same correct registration code as you... but in CommonProxy, instead of the main mod class. I have no idea where that function is actually called, and if it's called along with the entire CommonProxy - when that happens during loading, but I do believe it's caused by the code for BloodTNT registration not even running for client.
https://github.com/Arcaratus/BloodArsenal/blob/master/src/main/java/com/arc/bloodarsenal/common/CommonProxy.java#L31  

Additionally I think it's _uncovered_ by VoidMonster, as opposed to _caused_. BloodTNT fails to register on client even with just Blood Arsenal, and none of Reika's mods - there is no clientside entity visible for BloodTNT with or without DragonAPI.

**Comment by ReikaKalseki** [comment]
That sounds like a serious bug on their end.
